nvflare/app_common/abstract/fl_model.py

```diff
@@ -11,84 +11,87 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 from enum import Enum
-from typing import Dict, Optional
+from typing import Any, Dict, Optional
 
 
-class TransferType(Enum):
-    MODEL = "MODEL"
-    MODEL_DIFF = "MODEL_DIFF"
-    METRICS_ONLY = "METRICS_ONLY"
+class ParamsType(str, Enum):
+    FULL = "FULL"
+    DIFF = "DIFF"
 
 
 class FLModelConst:
-    AGGREGATION = "aggregation"
+    PARAMS_TYPE = "params_type"
+    PARAMS = "params"
+    OPTIMIZER_PARAMS = "optimizer_params"
     METRICS = "metrics"
+    CLIENT_WEIGHTS = "client_weights"
+    CURRENT_ROUND = "current_round"
+    TOTAL_ROUNDS = "total_rounds"
+    META = "meta"
+    AGGREGATION = "aggregation"
 
 
 class FLModel:
     def __init__(
         self,
-        transfer_type: TransferType,
-        model: Optional[Dict] = None,
-        optimizer: Optional[Dict] = None,
+        params_type: Optional[ParamsType] = None,
+        params: Any = None,
+        optimizer_params: Any = None,
         metrics: Optional[Dict] = None,
-        configs: Optional[Dict] = None,
         client_weights: Optional[Dict] = None,
-        round: Optional[int] = None,
+        current_round: Optional[int] = None,
+        total_rounds: Optional[int] = None,
         meta: Optional[Dict] = None,
     ):
         """
         Args:
-            transfer_type:
-                how the model will be transferred: as whole model (such as weight) or model_diff (weight_diff).
-                when transfer_type is METRICS_ONLY, model is None.
-            model:  machine learning model, for Deep learning, this could be weights or others depending on type of model
-            optimizer: optionally provided optimizer. For many cases, this optimizer doesn't need to be transferred during FL training.
+            params_type: type of the parameters. It only describes the "params".
+                In the case of "metrics", this params_type has no use.
```

Comment:
I think this is confusing and might not always be true. I suggest to delete "In the case of "metrics", this params_type has no use."

Reply:
thanks, fixed.